### PR TITLE
Kaggarwal/text gen onboard

### PR DESCRIPTION
**Pull request by aggarwal-k**

### CODE REVIEW

File: assets/training/finetune_acft_hf_nlp/components/finetune/text_generation/spec.yaml [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+      - steps
+    description: The logging strategy to adopt during training.
+
+  logging_steps:
+    type: integer
+    default: 500
+    optional: true
+    description: Number of update steps between two logs if logging_strategy='steps'
+
+  metric_for_best_model:
```
Comment (jpmann) [inline assets/training/finetune_acft_hf_nlp/components/finetune/text_generation/spec.yaml:243]: Is _bleu_ valid as part of _metric_for_best_model_?

**Comment by jpmann** [comment]
_eval_accumulation_steps _ is missing in readme.

File: assets/training/finetune_acft_hf_nlp/components/pipeline_components/text_generation/spec.yaml [hunk 2]
```diff
@@ -0,0 +1,592 @@
...
+      task: text-generation
+      test_data: '${{parent.jobs.text_generation_datapreprocess.outputs.output_dir}}'
+      label_column_name: '${{parent.inputs.ground_truth_key}}'
+      input_column_names: "'${{parent.inputs.text_key}}'"
+      batch_size: '${{parent.inputs.batch_size}}'
+      device: auto
+      mlflow_model: '${{parent.jobs.text_generation_finetune.outputs.mlflow_model_folder}}'
+  compute_metrics:
+    type: command
+    component: azureml://registries/azureml/components/compute_metrics/versions/0.0.10
```
Comment (skanakamedal) [inline assets/training/finetune_acft_hf_nlp/components/pipeline_components/text_generation/spec.yaml:578]: Shouldn't we upgrade to latest ME components?

File: assets/training/finetune_acft_hf_nlp/components/finetune/text_generation/spec.yaml (hunk 1 above)
Comment (aggarwal-k) [reply assets/training/finetune_acft_hf_nlp/components/finetune/text_generation/spec.yaml:237]: No its not, removed it

File: assets/training/finetune_acft_hf_nlp/components/finetune/text_generation/spec.yaml [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+name: text_generation_finetune
+version: 0.0.1
+type: command
+
+is_deterministic: True
+
+display_name: Text Generation Finetune
+description: Component to finetune model for Text Generation task
+
+environment: azureml:acft-contrib-hf-nlp-curated-acpt-pytorch-113-py38-cuda117-gpu:0.0.1
```
[the remark this answers is not in the document]
Comment (aggarwal-k) [reply assets/training/finetune_acft_hf_nlp/components/finetune/text_generation/spec.yaml]: yes, updated

**Review by skanakamedal — APPROVED** [review]

**Review by amangupta26 — APPROVED** [review]

**Review by jpmann — APPROVED** [review]

**Pull request merged**